test/pool_test.dart

```diff
@@ -105,6 +105,68 @@ void main() {
     });
   });
 
+  group("streamWithResource()", () {
+    test("The Stream holds a resource until it is completed.", () {
+      new FakeAsync().run((async) {
+        StreamController<int> streamController = new StreamController<int>();
+        var pool = new Pool(1);
+        pool.streamWithResource(() => streamController.stream);
+
+        bool nextTaskCompleted = false;
+        pool.withResource(() {
+          nextTaskCompleted = true;
+        });
+
+        new Future.delayed(new Duration(microseconds: 1), () {
+          expect(nextTaskCompleted, isFalse);
+          streamController.close();
+          return new Future.delayed(new Duration(microseconds: 1));
+        }).then((_) {
```

Comment:
The other tests were written before `async`/`await` existed, but new tests should use them in preference to `.then()`.